uaclient/entitlements/fips.py

```diff
@@ -25,10 +27,16 @@ class FIPSCommonEntitlement(repo.RepoEntitlement):
     def packages(self) -> "List[str]":
         packages = list(self.fips_required_packages)
         installed_packages = apt.get_installed_packages()
-        for pkg_name, extra_pkgs in self.fips_packages.items():
+
+        pkg_groups = groupby(
+            filter(lambda pkg: pkg not in packages, super().packages),
```

Comment:
I like this use of groupby, we probably can drop the filter so we don't have to perform that work for the filter out the pkg from fips_required_packages if we just append that required pkg name  when we are processing the pkg_groups below.

Reply:
Just realized that I forgot to remove the fips_package dict :facepalm: 
Thanks for the review @blackboxsw. I have updated the code